app/src/main/java/ru/vssemikoz/newsfeed/MainActivity.java

```diff
@@ -16,36 +18,35 @@
 import retrofit2.Response;
 import retrofit2.Retrofit;
 import retrofit2.converter.gson.GsonConverterFactory;
+import ru.vssemikoz.newsfeed.DAO.NewsItemDAO;
+import ru.vssemikoz.newsfeed.DataBase.NewsAppDataBase;
 import ru.vssemikoz.newsfeed.adapters.NewsFeedAdapter;
 import ru.vssemikoz.newsfeed.api.NewsApi;
 import ru.vssemikoz.newsfeed.models.NewsApiResponseItem;
 import ru.vssemikoz.newsfeed.models.NewsApiResponse;
 import ru.vssemikoz.newsfeed.models.NewsItem;
 
 public class MainActivity extends AppCompatActivity {
-    String KEY = "c94a57cbbb50497f94a2bb167dc91fc5";
-    String MAIN_URL = "https://newsapi.org";
-
+    MainApplication globalVariables;
+    NewsItemDAO newsItemDAO;
+    Callback<NewsApiResponse> callbackNewsItemList;
     List<NewsApiResponseItem> newsApiResponseItems = new ArrayList<>();
-    List<NewsItem> newsItemList = new ArrayList<>();
+    List<NewsItem> newsItemsFromDB = new ArrayList<>();
     NewsFeedAdapter adapter = new NewsFeedAdapter();
     RecyclerView recyclerView;
-    Callback<NewsApiResponse> callbackNewsItemList;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        globalVariables = (MainApplication) getApplicationContext();
 
         initRecView();
+        initDataBase();
```

Comment:
You're not initializing database here - you're loading entries. Please, call your methods in accordance with their responsibility/job.